### ISSUE TITLE
Plugin is displayed as Not Compatible with IntelliJ 2020.2.1

### ISSUE THREAD

**Issue by CC007** [mplushnikov/lombok-intellij-plugin#915]
### Short description
When trying to update IntelliJ, it complains that the lombok plugin is incompatible.

### Version information

 * **IDEA Version:** 2020.1.4 -> 2020.2.1
 * **OS Type & Version:** Windows 7 Enterprise (build 7601: service pack 1)
 * **Lombok Plugin Version:** 0.30-2020.1

### Steps to reproduce
1. From IntelliJ 2020.1.4: go to Help -> Check for updates...
2. In the popup click Update...

**Comment by lgathy** [comment]
It is compatible with 0.30-EAP version of the plugin, please try removing and re-installing the plugin after IDEA update to verify.

**Comment by mplushnikov** [comment]
There is a compatible version of plugin in plugin center. Just remove and install plugin again after update.
I'm in contact with market place admin to figure out, what is going wrong.

**Comment by CasparLi** [comment]
Resolved! Remove and Re-install then RESTART ide. Then,  everything goes well!

**Comment by ccamel** [comment]
Great. You saved my day! 🤗

**Comment by seenukarthi** [comment]
> There is a compatible version of plugin in plugin center. Just remove and install plugin again after update.
> I'm in contact with market place admin to figure out, what is going wrong.

Is this because of "EAP" in the version?

**Comment by aareed** [comment]
Just downloaded the latest version of IntelliJ 2020.2.1 (Ultimate Edition) and the Lombok plugun from Marketplace:  Lombok 0.30-EAP does not work.  First it just failed with a message about it being incompatable.  When I restarted IntelliJ and tried again, it hung after downloading 99%.  So there are two issues here:  1) The plugin is incompatible and 2) There seems to also be some problem downloading anything from https://plugins.jetbrains.com/plugin/6317-lombok/versions  For example it will download 553/548 KB and then stall.

**Comment by rwperrott** [comment]
Lombok 0.30-EAP, on marketplace too, is faulty for Refactor -> Delombok -> Constructors on the main java folder of a Maven project, because it causes this "IDE Fatal Errors" stacktrace:
```
java.lang.StackOverflowError
	at de.plushnikov.intellij.plugin.lombokconfig.ConfigIndexKey.hashCode(ConfigIndexKey.java:39)
	at de.plushnikov.intellij.plugin.lombokconfig.LombokConfigIndex$2.getHashCode(LombokConfigIndex.java:108)
	at de.plushnikov.intellij.plugin.lombokconfig.LombokConfigIndex$2.getHashCode(LombokConfigIndex.java:105)
	at com.intellij.util.containers.hash.HashUtil.hash(HashUtil.java:24)
	at com.intellij.util.containers.hash.LinkedHashMap.get(LinkedHashMap.java:88)
	at com.intellij.util.containers.SLRUMap.get(SLRUMap.java:80)
	at com.intellij.util.containers.SLRUCache.get(SLRUCache.java:42)
	at com.intellij.util.indexing.impl.MapIndexStorage.read(MapIndexStorage.java:214)
	at com.intellij.util.indexing.impl.MapIndexStorage.read(MapIndexStorage.java:37)
	at com.intellij.util.indexing.impl.storage.TransientChangesIndexStorage.read(TransientChangesIndexStorage.java:203)
	at com.intellij.util.indexing.impl.MapReduceIndex.getData(MapReduceIndex.java:213)
	at com.intellij.util.indexing.FileBasedIndexEx.lambda$processValueIterator$9(FileBasedIndexEx.java:319)
	at com.intellij.util.indexing.FileBasedIndexEx.lambda$processExceptions$4(FileBasedIndexEx.java:219)
	at com.intellij.util.ConcurrencyUtil.withLock(ConcurrencyUtil.java:228)
	at com.intellij.util.indexing.FileBasedIndexEx.lambda$processExceptions$5(FileBasedIndexEx.java:219)
	at com.intellij.util.indexing.IndexAccessValidator.validate(IndexAccessValidator.java:43)
	at com.intellij.util.indexing.FileBasedIndexEx.processExceptions(FileBasedIndexEx.java:219)
	at com.intellij.util.indexing.FileBasedIndexEx.processValueIterator(FileBasedIndexEx.java:318)
	at com.intellij.util.indexing.FileBasedIndexEx.processValuesInScope(FileBasedIndexEx.java:289)
	at com.intellij.util.indexing.FileBasedIndexEx.getValues(FileBasedIndexEx.java:96)
	at de.plushnikov.intellij.plugin.lombokconfig.ConfigDiscovery.readProperty(ConfigDiscovery.java:150)
	at de.plushnikov.intellij.plugin.lombokconfig.ConfigDiscovery.discoverProperty(ConfigDiscovery.java:119)
	at de.plushnikov.intellij.plugin.lombokconfig.ConfigDiscovery.getStringLombokConfigProperty(ConfigDiscovery.java:37)
	at de.plushnikov.intellij.plugin.lombokconfig.ConfigDiscovery.getBooleanLombokConfigProperty(ConfigDiscovery.java:44)
	at de.plushnikov.intellij.plugin.processor.modifier.FieldDefaultsModifierProcessor.isConfigDefaultFinal(FieldDefaultsModifierProcessor.java:101)
	at de.plushnikov.intellij.plugin.processor.modifier.FieldDefaultsModifierProcessor.canBeAffected(FieldDefaultsModifierProcessor.java:96)
	at de.plushnikov.intellij.plugin.processor.modifier.FieldDefaultsModifierProcessor.isSupported(FieldDefaultsModifierProcessor.java:49)
	at de.plushnikov.intellij.plugin.action.delombok.DelombokHandler.lambda$rebuildModifierList$1(DelombokHandler.java:115)
	at java.base/java.util.Arrays$ArrayList.forEach(Arrays.java:4390)
	at de.plushnikov.intellij.plugin.action.delombok.DelombokHandler.rebuildModifierList(DelombokHandler.java:114)
	at java.base/java.util.Arrays$ArrayList.forEach(Arrays.java:4390)
	at de.plushnikov.intellij.plugin.action.delombok.DelombokHandler.processModifierList(DelombokHandler.java:105)
	at de.plushnikov.intellij.plugin.action.delombok.DelombokHandler.invoke(DelombokHandler.java:58)
	at de.plushnikov.intellij.plugin.action.delombok.DelombokHandler.invoke(DelombokHandler.java:71)
        .... and loads more ...
	at de.plushnikov.intellij.plugin.action.delombok.DelombokHandler.invoke(DelombokHandler.java:71)
```

**Comment by libalex** [comment]
In a [marketplace](https://plugins.jetbrains.com/plugin/6317-lombok/), Lombok plugin displayed as "Compatible with all IntelliJ-based IDEs".
I using IDEA 2020.2.1 - and Lombok plugin correctly installing and work.
I tried update IDEA 2020.2 with Lombok plugin to IDEA 2020.2.1 and clean install IDEA 2020.2.1 and Lombok plugin - all is work.

**Comment by CC007** [comment]
For me lombok complains about being incompatible, but once I updated, it automatically installed the EAP version

**Comment by libalex** [comment]
Strange, in my case Lombok plugin was updated to version 0.30-EAP even before IDEA was updated

**Comment by CC007** [comment]
The difference probably was that I came from 2020.1.x

**Comment by mplushnikov** [comment]
This is a problem of IntelliJ https://youtrack.jetbrains.com/issue/IDEA-246822, and it should be fixed in the next release (Included in builds: 203.3261) of IntelliJ. (https://github.com/JetBrains/intellij-community/commit/e647fceb25ec831371949a6aacf7685e036dba7a)

At the moment you can only do a workaround by uninstalling and then installing the plugin again.
